cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/core/mongo/impl/PublicationDaoImpl.java

```diff
@@ -68,47 +72,83 @@
         return results;
     }
 
+    @Override
+    public List<PublicationImpl> getPublications(Pageable pageable, PublicationImpl pub, boolean init) {
+
+        String[] title = pageable.getSort().toString().split(": ");
+        int limit = pageable.getPageSize();
+        Criteria criteria1 = null;
+        Criteria criteria2 = null;
+        if (!init) {
+            if (title[0].equals(SortType.publishYear.toString())) {
+                criteria1 = Criteria.where(title[0]).gt(pub.getPublishYear());
+                criteria2 = Criteria.where(title[0]).is(pub.getPublishYear());
+            } else if (title[0].equals(SortType.journal.toString())) {
+                criteria1 = Criteria.where(title[0]).gt(pub.getJournal());
+                criteria2 = Criteria.where(title[0]).is(pub.getJournal());
+            } else {
+                criteria1 = Criteria.where(title[0]).gt(pub.getMetadata().getTitle());
+                criteria2 = Criteria.where(title[0]).is(pub.getMetadata().getTitle());
+            }
```

Comment:
these huge if/else blocks are not very maintainable. What if we want to add more options for sorting? I would try to maybe just have a map that maps fields to methods that are then called.